### ISSUE TITLE
Remove Block Option Not Intuitive With Screen Readers

### ISSUE THREAD

**Issue by donalirl** [WordPress/gutenberg#22012]
**Describe the bug**
A visually-impaired user using a screen reader cannot find the Remove Block option because it is buried.

Using a screen reader like Apple Voiceover, when you tap on block to get to the settings, the settings expand but do nothing as far as a screen reader is concerned. This is because the settings expand to the left, which a visually-impaired user cannot find because they navigate left to right. It's not at all intuitive to go backwards. It should be possible to navigate forwards to find the block settings to remove a block.

**To reproduce**
1. Edit any page or post that has blocks on it
2. Navigate using a screen reader to select any block
3. Tap on the block settings
4. Attempt to find the Remove Block option

**Expected behavior**
Using a screen reader, I expected to be able to navigate through my blocks and remove a block.

**Editor version (please complete the following information):**
WordPress version: 5.4
Does the website has Gutenberg plugin installed, or is it using the block editor that comes by default? Default

**Desktop (please complete the following information):**
Safari using Apple Voiceover

**Comment by getsource** [comment]
Hi! Thanks much for the report, and sorry it's taken so long to get a reply.

A couple of folks tested this [during the Gutenberg triage today](https://wordpress.slack.com/archives/C02QB2JS7/p1601358157083200), and found that it was being announced with Voiceover on MacOS (this was not tested in the meeting with iOS).

Since there wasn't anyone from the Accessibility team present at the meeting, I'm reaching out for any additional feedback.

If in your testing it looks like this has been fixed since the report, that'd be great to know as well.

Thanks again!

**Comment by donalirl** [comment]
@getsource No worries! I appreciate you looking into this.

I don't have access to Apple devices anymore, so I tested on my Android (version 10) with latest version of Chrome app. I could not get to the block settings at all this time. 

I recorded myself fumbling with Android voiceover here: https://d.pr/v/2LoJm0 You can see I have a paragraph block and an image block, but can't swipe left or right with Android voice assistance to get to the block toolbar.

**Comment by afercia** [comment]
@getsource thanks for the ping on Slack. Not sure testing on macOS as done during the Gutenberg triage makes much sense as this needs to be tested on iOS using gestures instead of the keyboard. Sure, the fact this issue is related to mobile devices wasn't cleat to me as well after reading the first description.

@donalirl thanks for your report! I'm not sure I understand what you meant by "the settings expand to the left". If you meant that the block toolbar itself appears _before_ the block then yes, that's how it works. Once the toolbar is displayed, the "More options" button in the toolbar opens a menu which contains the button to remove the block. I agree this is buried down in a menu that requires lots of steps to navigate to.

Additionally, on small screens, the block toolbar is, I would say unfortunately, moved to the top and becomes "fixed". This means that the block toolbar is actually very far from the block itself. This might explain why a single swipe left doesn't navigate to the toolbar. I'm afraid that on mobile there's the need to swipe left multiple times to get to the toolbar.

If I missed something, please do feel free to add more detailed steps to reproduce.

Unfortunately, the blocks accessibility on mobile is far from ideal. There are anti-patterns that the accessibility team has pointed to several times like:
- continuous appearing and disappearing of parts of the user interface, e.g. the block toolbar
- visual order and DOM order mismatch
- non-ideal tab sequence
- etc.

There's still a lot of work to do 🙂 For now, I'd recommend to test and reproduce this on iOS VoiceOver. I'm afraid I can't test myself, as I don't have access to a Apple mobile device.

**Comment by enriquesanchez** [comment]
I tested with VoiceOver on iOS 14.

The first try I was able to reach the block settings options and the block inspector. That said, as @afercia mentions, it required quite a bit of swipes to get there, and the focus order did not make any sense. 

**Accessing the "More options" menu in the block's toolbar:**

![IMG_4174](https://user-images.githubusercontent.com/3276087/94577822-9bcb0900-023c-11eb-8905-1db55a31a479.PNG)

**Navigating the list of options in the "More options" menu:**

![IMG_4175](https://user-images.githubusercontent.com/3276087/94577834-9e2d6300-023c-11eb-84d9-7748e0de7343.PNG)

**Accessing the block inspector:**

![IMG_4176](https://user-images.githubusercontent.com/3276087/94577836-9ec5f980-023c-11eb-888e-186fd4ccec6a.PNG)

Unfortunately on a second and third attempt **I was not able to reach any of the items inside the "More options" menu**. I was not able to swipe into the menu and my focus kept going back to the editor toolbar:

![IMG_4177](https://user-images.githubusercontent.com/3276087/94579009-f7e25d00-023d-11eb-80a5-4410d8a9d6c9.PNG)

I'm not sure what changed between this and my first attempt. I know I didn't do anything different, so it might well be a VoiceOver quirk.

>when you tap on block to get to the settings, the settings expand but do nothing as far as a screen reader is concerned

I think this confirms this report. VoiceOver's focus cannot open the "More options" menu and therefore the "Remove block" option is unrechable.

**Comment by afercia** [comment]
I wish testing on mobile was easier 😆 Thanks @enriquesanchez !

**Comment by inaikem** [comment]
@afercia, @enriquesanchez, it's me again 👋 Just checking in to see if any progress has been made since your comment on mobile testing in September 2020. Thanks!
